File: api/auto-score.js
import Anthropic from '@anthropic-ai/sdk';

export default async function handler(req, res) {
  // CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    // Check API key
    const apiKey = process.env.ANTHROPIC_API_KEY;
    if (!apiKey) {
      return res.status(500).json({ error: 'ANTHROPIC_API_KEY not configured' });
    }

    // Initialize client inside handler
    const client = new Anthropic({ apiKey });

    // Parse JSON body
    const payload = typeof req.body === 'string' ? JSON.parse(req.body) : req.body;
    const { company_name, website, industry, stage, geography, description } = payload;

    if (!company_name) {
      return res.status(400).json({ error: 'company_name is required' });
    }

    const prompt = `You are an expert business analyst. Analyze "${company_name}"${industry ? `, in ${industry}` : ''}${stage ? ` at ${stage}` : ''}.

Score across 7 SMEAT dimensions (Customer, People, Operations, Finance, Analytics, Risk, Impact).

Return ONLY valid JSON (no markdown, no explanation) matching this exact structure:
{
  "company_name": "company name",
  "summary": "2-3 sentence executive summary of the assessment",
  "segments": {
    "customer": {
      "subs": [
        {"name": "subsegment name", "maturity": 1, "impact": 1}
      ],
      "confidence": 0.85
    },
    ...repeat for each dimension...
  }
}

For maturity: 1=Advanced, 2=Developing, 3=Emerging, 4=Nascent
For impact: 1=Critical, 2=Neutral, 3=Low, 4=Not Needed
Each dimension should have 2-3 subsegments.`;

    res.setHeader('Content-Type', 'text/event-stream');
    res.setHeader('Cache-Control', 'no-cache');
    res.setHeader('Connection', 'keep-alive');

    const stream = await client.messages.create({
      model: 'claude-sonnet-4-6',
      max_tokens: 2048,
      messages: [{ role: 'user', content: prompt }],
      stream: true,
    });

    for await (const event of stream) {
      if (event.type === 'content_block_delta' && event.delta.type === 'text_delta') {
        res.write(`data: ${JSON.stringify({ type: 'chunk', data: event.delta.text })}\n\n`);
      }
    }

    res.end();
  } catch (error) {
    console.error('Error:', error);
    res.status(500).json({ error: error.message });
  }
}

File: api/auto_score.py
import json
from http.server import BaseHTTPRequestHandler
from urllib.parse import parse_qs
import os
from anthropic import Anthropic


class handler(BaseHTTPRequestHandler):
    def do_POST(self):
        """Handle POST requests to /api/auto-score"""
        try:
            content_length = int(self.headers.get('Content-Length', 0))
            body = self.rfile.read(content_length).decode('utf-8')

            # Parse form data
            data = parse_qs(body)
            company_name = data.get('company_name', [''])[0]
            website = data.get('website', [''])[0]
            industry = data.get('industry', [''])[0]
            stage = data.get('stage', [''])[0]
            geography = data.get('geography', [''])[0]
            description = data.get('description', [''])[0]

            if not company_name:
                self.send_response(400)
                self.send_header('Content-type', 'text/plain')
                self.end_headers()
                self.wfile.write(b'company_name is required')
                return

            # Build prompt
            prompt = f"""You are an expert business analyst. Analyze "{company_name}"{f", in {industry}" if industry else ""}{f" at {stage}" if stage else ""}.

Score across 7 SMEAT dimensions (Customer, People, Operations, Finance, Analytics, Risk, Impact).
For each, rate maturity (1=Advanced to 4=Nascent) and impact (1=Critical to 4=Not needed).

Return ONLY valid JSON with the scores."""

            # Call Claude API
            api_key = os.getenv('ANTHROPIC_API_KEY')
            if not api_key:
                self.send_response(500)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({"error": "API key not configured"}).encode())
                return

            client = Anthropic(api_key=api_key)

            # Stream response
            self.send_response(200)
            self.send_header('Content-type', 'text/event-stream')
            self.send_header('Cache-Control', 'no-cache')
            self.send_header('Connection', 'keep-alive')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()

            with client.messages.stream(
                model="claude-3-5-sonnet-20241022",
                max_tokens=2048,
                messages=[{"role": "user", "content": prompt}],
            ) as stream:
                for text in stream.text_stream:
                    response = f"data: {json.dumps({'type': 'chunk', 'data': text})}\n\n"
                    self.wfile.write(response.encode())

        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({"error": str(e)}).encode())

    def do_OPTIONS(self):
        """Handle CORS preflight"""
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
